### PR TITLE
Various cleanup in plexus

### PR DESCRIPTION
**Pull request by tiffon**
## Which problem is this PR solving?

Various cleanup of the plexus package.

## Short description of the changes

**Note:** This builds on the #425 (the README PR).

- Revisit the default types for generics on `TEdge` and `TVertex`
- Remove the label field from `TVertex`
- Remove the data and label fields from `TEdge`
- Rename component to `Digraph` instead of `LayeredDigraph`
- Add props factories onto `Digraph` as static fields
- Rename `getLocalId` to `getGlobalId`
- Remove `scaledStrokeWidth` prop factory
- Fix perf refressions in `Digraph` by wrapping nodes and edges in pure wrappers
- Fix issues in jaeger-ui introduced when removing `number` from `TVertexKey`

### CODE REVIEW

File: packages/plexus/src/LayoutManager/LayoutManager.tsx [hunk 1]
```diff
@@ -17,17 +17,17 @@ import { TCancelled, TEdge, TLayoutDone, TPendingLayoutResult, TPositionsDone, T
...
+  resolveLayout?: (result: TCancelled | TLayoutDone<T, U>) => void;
 };
 
-export default class LayoutManager<T = {}> {
+export default class LayoutManager {
   layoutId: number;
-  coordinator: Coordinator<T>;
-  pendingResult: TPendingResult<T> | null;
+  coordinator: Coordinator;
+  pendingResult: TPendingResult<any, any> | null;
```
Comment (everett980) [inline packages/plexus/src/LayoutManager/LayoutManager.tsx]: why were the generics here and in `_handleUpdate` replaced with `any`? does it simply not matter for LayoutManager?

File: packages/plexus/src/LayoutManager/match-inputs.tsx [hunk 2]
```diff
@@ -14,8 +14,11 @@
 
 import { TEdge, TLayoutEdge, TLayoutVertex, TSizeVertex, TVertex } from '../types';
 
-export function matchEdges(input: TEdge[], output: TLayoutEdge[]): TLayoutEdge[] {
-  const map: { [key: string]: TEdge } = {};
+export function matchEdges<T = Record<string, unknown>>(
+  input: TEdge<T>[],
+  output: TLayoutEdge<{}>[]
+): TLayoutEdge<T>[] {
+  const map: { [key: string]: TEdge<any> } = {};
```
Comment (everett980) [inline packages/plexus/src/LayoutManager/match-inputs.tsx]: I think `any` can be replaced with `T`, right?

File: packages/plexus/src/zoom/MiniMap.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ export function MiniMap(props: TProps) {
       <div className={`${css.item} ${css.map}`} style={mapSize}>
         <div className={css.mapActive} style={{ ...activeXform, ...mapSize }} />
       </div>
-      <div className={`${css.item} ${css.button}`} onClick={props.viewAll} role="button">
+      <div className={`${css.item} ${css.button}`} onClick={props.viewAll} data-wtf="true" role="button">
```
Comment (everett980) [inline packages/plexus/src/zoom/MiniMap.tsx]: is this a holdover from testing?

Comment (tiffon) [reply packages/plexus/src/zoom/MiniMap.tsx]: lol, doh!!

File: packages/plexus/src/DirectedGraph/DirectedGraph.tsx [hunk 4]
```diff
@@ -74,7 +73,7 @@ export default class DirectedGraph<T> extends React.PureComponent<
     arrowScaleDampener: undefined,
     className: '',
     classNamePrefix: 'plexus',
-    getNodeLabel: defaultGetNodeLabel,
+    // getNodeLabel: defaultGetNodeLabel,
```
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/plexus/src/DirectedGraph/DirectedGraph.tsx]: Yes, the entire `DirectedGraph` component will be removed in another PR.

File: packages/plexus/src/LayoutManager/Coordinator.tsx [hunk 5]
```diff
@@ -250,14 +255,15 @@ export default class Coordinator<T> {
     }
     // phase is either edges or dot-only
     if (edges) {
-      const adjEdgeCoords = convCoord.edgeToPixels.bind(null, graph);
-      let adjEdges = edges.map<TLayoutEdge>(adjEdgeCoords);
-      adjEdges = matchEdges(input.edges, adjEdges);
+      // const adjEdgeCoords = convCoord.edgeToPixels<{}>.bind(null, graph);
+      // let adjEdges = edges.map((edge: TEdge<{}>) => convCoord.edgeToPixels(graph, edge) adjEdgeCoords);
```
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/plexus/src/LayoutManager/Coordinator.tsx]: Thanks.

File: packages/plexus/src/DirectedGraph/DirectedGraph.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/plexus/src/DirectedGraph/DirectedGraph.tsx]: Will resolve this in the PR that removes `DirectedGraph`

File: packages/plexus/src/LayoutManager/LayoutManager.tsx (hunk 1 above)
Comment (tiffon) [reply packages/plexus/src/LayoutManager/LayoutManager.tsx]: Correct, it's not relevant to the `LayoutManager` which only needs to work with the data on `TEdge` and `TSizeVertex`.

**Comment by tiffon** [comment]
I think the generics need a bit more refinement. I suggest this happen on a future PR as the generics are currently in pretty good shape and we have other fish to fry. Currently, the main place to supply types is on the `Digraph` component, everything else trickles down, from there.

**Pull request merged**